src/ModelIO/MDLAnimatedValueTypes.cs

```diff
@@ -0,0 +1,693 @@
+				fixed (double* timesPtr = times)
+					_ResetWithDoubleArray ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual float [] GetFloatValues ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new float [(int) count];
+
+			unsafe {
+				fixed (float* ptr = timesArr)
+					_GetFloatArray ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+
+		public virtual double [] GetDoubleValues ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new double [(int) count];
+
+			unsafe {
+				fixed (double* ptr = timesArr)
+					_GetDoubleArray ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+	}
+
+	public partial class MDLAnimatedVector2 {
+
+		public virtual void Reset (Vector2 [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Vector2* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithFloat2Array ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual void Reset (Vector2d [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Vector2d* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithDouble2Array ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual Vector2 [] GetVector2Values ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new Vector2 [(int) count];
+
+			unsafe {
+				fixed (Vector2* ptr = timesArr)
+					_GetFloat2Array ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+
+		public virtual Vector2d [] GetVector2dValues ()
+		{
+			var count = TimeSampleCount;
+			var timesArr = new Vector2d [(int) count];
+
+			unsafe {
+				fixed (Vector2d* ptr = timesArr)
+					_GetDouble2Array ((IntPtr) ptr, count);
+			}
+			return timesArr;
+		}
+	}
+
+	public partial class MDLAnimatedVector3 {
+
+		public virtual void Reset (Vector3 [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Vector3* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithFloat3Array ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
+			}
+		}
+
+		public virtual void Reset (Vector3d [] values, double [] times)
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Vector3d* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithDouble3Array ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
```

Comment:
What's the length of the `values` array supposed to be? Identical to the length of `times`? If so, we should add checks for it.